NebulaNetwork/SaveManager.cs

```diff
@@ -119,43 +125,45 @@ public static void LoadServerData()
                 // Supported revision: 5~8
                 if (revision is < 5 or > REVISION)
                 {
-                    throw new Exception();
+                    throw new Exception($"Unsupport version {revision}");
                 }
             }
-        }
-        catch (Exception)
-        {
-            Log.Warn("Skipping server data from unsupported Nebula version...");
-            return;
-        }
 
-        var playerNum = netDataReader.GetInt();
+            var playerNum = netDataReader.GetInt();
 
 
-        for (var i = 0; i < playerNum; i++)
-        {
-            var hash = netDataReader.GetString();
-            PlayerData playerData = null;
-            switch (revision)
+            for (var i = 0; i < playerNum; i++)
             {
-                case REVISION:
-                    playerData = netDataReader.Get(() => new PlayerData());
-                    break;
-                case >= 5:
-                    playerData = new PlayerData();
-                    playerData.Import(netDataReader, revision);
-                    break;
-            }
+                var hash = netDataReader.GetString();
+                PlayerData playerData = null;
+                switch (revision)
+                {
+                    case REVISION:
+                        playerData = netDataReader.Get(() => new PlayerData());
+                        break;
+                    case >= 5:
+                        playerData = new PlayerData();
+                        playerData.Import(netDataReader, revision);
+                        break;
+                }
 
-            if (!playerSaves.ContainsKey(hash) && playerData != null)
-            {
-                playerSaves.Add(hash, playerData);
-            }
-            else if (playerData == null)
-            {
-                Log.Warn($"could not load player data from unsupported save file revision {revision}");
+                if (!playerSaves.ContainsKey(hash) && playerData != null)
+                {
+                    playerSaves.Add(hash, playerData);
+                }
+                else if (playerData == null)
+                {
+                    Log.Warn($"Could not load player data from unsupported save file revision {revision}");
+                }
             }
         }
+        catch (Exception e)
+        {
+            playerSaves.Clear();
+            Log.WarnInform("Skipping server data due to following excpetion:\n" + e.Message);
```

Comment:
```suggestion
            Log.WarnInform("Skipping server data due to exception:\n" + e.Message);
```
or "the following exception" but maybe that's too long